### PR TITLE
Create a build script for XCFramework

### PR DESCRIPTION
**Pull request by vvechkanov**
Things to consider before you submit the PR:

* [ ] ~~Has `CHANGELOG.md` been updated?~~ (This is part of the feature, docs will be changed later)
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] ~~Did you add unit tests?~~ (only script changes)
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Create a build script for XCFramework.

## Related PRs or issues

[AB#79026](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/79026)

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: Scripts/build-xcframework.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Install dir will be the final output to the framework.
+# The following line create it in the root folder of the current project.
+PRODUCTS_DIR="${SRCROOT}/../AppCenter-SDK-Apple/"
+WORK_DIR="${PRODUCTS_DIR}/xcframework/"
+
+rm -rf ${PROJECT_NAME}.xcframework/
+xcodebuild -create-xcframework -framework $WORK_DIR/Release-iphoneos/${PROJECT_NAME}.framework -framework $WORK_DIR/Release-iphonesimulator/${PROJECT_NAME}.framework -framework $WORK_DIR/Release-appletvos/${PROJECT_NAME}.framework -framework $WORK_DIR/Release-appletvsimulator/${PROJECT_NAME}.framework -framework $WORK_DIR/Release-macos/${PROJECT_NAME}.framework -output "$WORK_DIR/Output/${PROJECT_NAME}.xcframework"
```
Comment (annakocheshkova) [inline Scripts/build-xcframework.sh:12]: Consider adding separate variables for device/simulator/macos dirs.

File: Scripts/build-macos-framework.sh [hunk 2]
```diff
@@ -36,3 +39,7 @@ mkdir -p "${PRODUCTS_DIR}"
 
 # Copy framework.
 cp -R "${DEVICE_DIR}/${PROJECT_NAME}.framework" "${PRODUCTS_DIR}"
+
+mkdir -p "${XCFRAMEWORK_DIR}/Release-macos/"
```
Comment (annakocheshkova) [inline Scripts/build-macos-framework.sh:43]: Consider adding a variable for this path as well

File: AppCenter/AppCenter.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -2656,7 +2683,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "\"${SRCROOT}/../Scripts/VerifyNoBuildSettings.swift\" ${PROJECT_NAME}.xcodeproj/project.pbxproj\n";
+			shellScript = "#${SRCROOT}/../Scripts/VerifyNoBuildSettings.swift\" ${PROJECT_NAME}.xcodeproj/project.pbxproj\n";
```
Comment (annakocheshkova) [inline AppCenter/AppCenter.xcodeproj/project.pbxproj:2686]: Is this expected change?

Comment (vvechkanov) [reply AppCenter/AppCenter.xcodeproj/project.pbxproj:2686]: Nope, thank you! I will fix it.

File: Scripts/build-xcframework.sh (hunk 1 above)
Comment (vvechkanov) [reply Scripts/build-xcframework.sh]: I rewrite this script a bit. Pls rereview

File: Scripts/build-xcframework.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+# Clean previus XCFramework build.
+rm -rf ${PROJECT_NAME}.xcframework/
+
+# Build XCFramework.
+XC_BUILD_COMMAND="xcodebuild -create-xcframework"
+
+# Create a cycle instead next lines
+PLATFORM_NAME="iphoneos"
+FRAMEWORK_PATH="$WORK_DIR/Release-${PLATFORM_NAME}/${PROJECT_NAME}.framework"
+[ -e "$FRAMEWORK_PATH" ] && XC_BUILD_COMMAND="$XC_BUILD_COMMAND -framework $FRAMEWORK_PATH"
```
Comment (MatkovIvan) [inline Scripts/build-xcframework.sh:21]: Let's replace `XC_BUILD_COMMAND` to just function

File: Scripts/build-tvos-framework.sh [hunk 5]
```diff
@@ -39,5 +42,9 @@ mkdir -p "${PRODUCTS_DIR}"
 # Copy framework.
 cp -R "${DEVICE_DIR}/${PROJECT_NAME}.framework" "${PRODUCTS_DIR}"
 
+mkdir -p "${XCFRAMEWORK_DIR}"
+
+cp -R "${WORK_DIR}/" "${XCFRAMEWORK_DIR}"
```
Comment (MatkovIvan) [inline Scripts/build-tvos-framework.sh:47]: Comments? Why do we need it?

File: Config/Global.xcconfig [hunk 6]
```diff
@@ -168,3 +168,6 @@ GCC_TREAT_WARNINGS_AS_ERRORS = YES
 
 // Swift Compiler - Code Generation
 SWIFT_ENFORCE_EXCLUSIVE_ACCESS = on
+
+// Option for xcframework
+BUILD_LIBRARIES_FOR_DISTRIBUTION=YES
```
Comment (MatkovIvan) [inline Config/Global.xcconfig:173]: Align format

File: Scripts/build-xcframework.sh [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+OUTPUT="${WORK_DIR}/Output"
+
+# Clean previus XCFramework build.
+rm -rf ${PROJECT_NAME}.xcframework/
+
+# Build XCFramework.
+XC_BUILD_COMMAND="xcodebuild -create-xcframework"
+
+# Create a cycle instead next lines
+PLATFORM_NAME="iphoneos"
```
Comment (annakocheshkova) [inline Scripts/build-xcframework.sh:19]: Please additionally move `AppCenterDistributeResources.bundle` under framework

File: Scripts/build-xcframework.sh [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+WORK_DIR="${SRCROOT}/../AppCenter-SDK-Apple/xcframework"
+
+# Work dir will be the final output to the framework.
+OUTPUT="${WORK_DIR}/Output"
+
+# Clean previus XCFramework build.
+rm -rf ${PROJECT_NAME}.xcframework/
+
+# Build XCFramework.
+XC_BUILD_COMMAND="xcodebuild -create-xcframework"
```
Comment (MatkovIvan) [inline Scripts/build-xcframework.sh:16]: Remove

File: Scripts/build-xcframework.sh [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+function  SET_XC_BUILD_COMMAND_FRAMEWORK() {
```
Comment (MatkovIvan) [inline Scripts/build-xcframework.sh:18]: Double space, rename to function style

File: Scripts/build-xcframework.sh (hunk 4 above)
Comment (olmisa) [reply Scripts/build-xcframework.sh]: Fixed

File: Scripts/build-xcframework.sh (hunk 8 above)
Comment (olmisa) [reply Scripts/build-xcframework.sh]: Fixed

File: Scripts/build-xcframework.sh (hunk 9 above)
Comment (olmisa) [reply Scripts/build-xcframework.sh]: Fixed

File: Scripts/build-xcframework.sh [hunk 10]
```diff
@@ -13,21 +13,19 @@ OUTPUT="${WORK_DIR}/Output"
...
-SET_XC_BUILD_COMMAND_FRAMEWORK "appletvsimulator"
-SET_XC_BUILD_COMMAND_FRAMEWORK "macos"
+SetXcBuildCommandFramework "iphoneos"
+SetXcBuildCommandFramework "iphonesimulator"
+SetXcBuildCommandFramework "appletvos"
+SetXcBuildCommandFramework "appletvsimulator"
+SetXcBuildCommandFramework "macos"
 
-XC_BUILD_COMMAND="$XC_BUILD_COMMAND -output  $OUTPUT/${PROJECT_NAME}.xcframework"
+XC_BUILD_COMMAND="xcodebuild -create-xcframework $XC_BUILD_COMMAND -output  $OUTPUT/${PROJECT_NAME}.xcframework"
```
Comment (MatkovIvan) [inline Scripts/build-xcframework.sh:28]: Double space. Please verify formatting in the changes

**Pull request merged**